packages/contracts-bedrock/contracts/test/invariants/OptimismPortal.t.sol

```diff
@@ -1,8 +1,39 @@
 pragma solidity 0.8.15;
 
+import { OptimismPortal } from "../../L1/OptimismPortal.sol";
+import { L2OutputOracle } from "../../L1/L2OutputOracle.sol";
+import { AddressAliasHelper } from "../../vendor/AddressAliasHelper.sol";
+import { SystemConfig } from "../../L1/SystemConfig.sol";
+import { ResourceMetering } from "../../L1/ResourceMetering.sol";
+import { Constants } from "../../libraries/Constants.sol";
+
 import { Portal_Initializer } from "../CommonTest.t.sol";
 import { Types } from "../../libraries/Types.sol";
 
+contract OptimismPortal_Depositor {
+    OptimismPortal internal portal;
+    bool public failedToComplete;
+
+    constructor(OptimismPortal _portal) {
+        portal = _portal;
+    }
+
+    // A test intended to identify any unexpected halting conditions
+    function depositTransactionCompletes(
+        address _to,
+        uint256 _mint,
+        uint256 _value,
+        uint64 _gasLimit,
+        bool _isCreation,
+        bytes memory _data
+    ) public payable {
+        failedToComplete = true;
+        require(!_isCreation || _to == address(0), "OptimismPortal_Depositor: invalid test case.");
+        portal.depositTransaction{ value: _mint }(_to, _value, _gasLimit, _isCreation, _data);
+        failedToComplete = false;
```

Comment:
Because `portal.depositTransaction` is checked by default, `actor.failedToComplete` will always equal false. Forge's invariant fuzzer performs several separate calls on each run to the `depositTransactionCompletes` function in the actor, and then checks the invariant afterwards in a separate call. To make sure the `failedToComplete = true` persists, we'll need to surround the call in a `try/catch`
```suggestion
        require(!_isCreation || _to == address(0), "OptimismPortal_Depositor: invalid test case.");
        try portal.depositTransaction{ value: _mint }(_to, _value, _gasLimit, _isCreation, _data) {
            // Do nothing; Call succeeded
        } catch {
            failedToComplete = true;
        }
```

Reply:
> Because `portal.depositTransaction` is checked by default, `actor.failedToComplete` will always equal false. Forge's invariant fuzzer performs several separate calls on each run to the `depositTransactionCompletes` function in the actor, and then checks the invariant afterwards in a separate call. To make sure the `failedToComplete = true` persists, we'll need to surround the call in a `try/catch`

Thanks for a great feedback!! I note that I ,apart from your suggestion, have added the bounding logic for fuzzing inputs in order to make 'forge test' passes.